manager/integration/tests/test_backing_image.py

```diff
@@ -0,0 +1,263 @@
+from common import csi_pv, pvc, pod_make  # NOQA
+from common import csi_pv_backingimage, pvc_backingimage  # NOQA
+from common import disable_auto_salvage  # NOQA
+
+from test_basic import volume_basic_test, volume_iscsi_basic_test,\
+    snapshot_test, backup_test, backup_labels_test
+from test_engine_upgrade import engine_offline_upgrade_test, \
+    engine_live_upgrade_test, engine_live_upgrade_rollback_test
+from test_ha import ha_simple_recovery_test, ha_salvage_test, \
+    ha_backup_deletion_recovery_test
+from test_csi import csi_mount_test, csi_io_test, csi_backup_test
+from test_recurring_job import recurring_job_labels_test
+
+from common import get_self_host_id
+from common import create_and_check_volume, wait_for_volume_healthy, \
+    wait_for_volume_delete, cleanup_all_volumes
+from common import create_backing_image_with_matching_url, \
+    wait_for_backing_image_disk_cleanup, cleanup_all_backing_images
+
+from common import BACKING_IMAGE_NAME, BACKING_IMAGE_QCOW2_URL, \
+    BACKING_IMAGE_RAW_URL, BACKING_IMAGE_EXT4_SIZE
+
+
+@pytest.mark.coretest   # NOQA
+@pytest.mark.backing_image  # NOQA
+def test_backing_image_basic_operation(client, volume_name):  # NOQA
+    for bi_url in (BACKING_IMAGE_QCOW2_URL, BACKING_IMAGE_RAW_URL):
+        create_backing_image_with_matching_url(
+            client, BACKING_IMAGE_NAME, bi_url)
+        backing_image_basic_operation_test(
+            client, volume_name, BACKING_IMAGE_NAME, bi_url)
+        cleanup_all_volumes(client)
+        cleanup_all_backing_images(client)
+
+
+def backing_image_basic_operation_test(client, volume_name, bi_name, bi_url):  # NOQA
+    """
+    Test Backing Image APIs.
+
+    1. Create a backing image.
+    2. Create and attach a Volume with the backing image set.
+    3. Verify that the all disk states in the backing image are "downloaded".
+    4. Try to use the API to manually clean up one disk for the backing image
+       but get failed.
+    5. Try to use the API to directly delete the backing image
+       but get failed.
+    6. Delete the volume.
+    7. Use the API to manually clean up one disk for the backing image
+    8. Delete the backing image.
+    """
+
+    volume = create_and_check_volume(
+        client, volume_name, 3,
+        str(BACKING_IMAGE_EXT4_SIZE), bi_name)
+    lht_host_id = get_self_host_id()
+    volume.attach(hostId=lht_host_id)
+    volume = wait_for_volume_healthy(client, volume_name)
+    assert volume.backingImage == bi_name
+    assert volume.size == str(BACKING_IMAGE_EXT4_SIZE)
+
+    random_disk_id = ""
+    backing_image = client.by_id_backing_image(bi_name)
+    assert backing_image.imageURL == bi_url
+    assert not backing_image.deletionTimestamp
+    assert len(backing_image.diskStateMap) == 3
+    for disk_id, state in iter(backing_image.diskStateMap.items()):
+        assert state == "downloaded"
+        random_disk_id = disk_id
+    assert random_disk_id != ''
+
```

Comment:
I think test is missing step 4 here ?

Reply:
Similarly the step 7 is missing, I don't see the disk cleanup step.

Reply:
Step 4 is this part, which is not missed:
```
    with pytest.raises(Exception):
        backing_image.backingImageCleanup(diks=[random_disk_id])
```

But in fact, step 5 will be skipped if when the operation in step4 is failed. I will update this part.

Reply:
Step 7 was the last line `test_backing_image_basic_operation(client, volume_name)`, which may be not straightforward. I will add a separate backing image deletion call at the end of **this** function.